homeassistant/components/seventeentrack/coordinator.py

```diff
@@ -0,0 +1,143 @@
+"""Coordinator for 17Track."""
+
+from dataclasses import dataclass
+from typing import Any
+
+from py17track import Client as SeventeenTrackClient
+from py17track.errors import SeventeenTrackError
+from py17track.package import Package
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_FRIENDLY_NAME, ATTR_LOCATION
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import slugify
+
+from .const import (
+    ATTR_DESTINATION_COUNTRY,
+    ATTR_INFO_TEXT,
+    ATTR_ORIGIN_COUNTRY,
+    ATTR_PACKAGE_TYPE,
+    ATTR_STATUS,
+    ATTR_TIMESTAMP,
+    ATTR_TRACKING_INFO_LANGUAGE,
+    ATTR_TRACKING_NUMBER,
+    CONF_SHOW_ARCHIVED,
+    CONF_SHOW_DELIVERED,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    LOGGER,
+)
+
+
+@dataclass
+class SeventeenTrackData:
+    """Class for handling the data retrieval."""
+
+    def __init__(self) -> None:
+        """Initialize the data object."""
+        self.summary: dict[str, dict[str, Any]] = {}
+        self.current_packages: dict[str, dict[str, Any]] = {}
+        self.new_packages: dict[str, Package] = {}
+        self.old_packages: set[Package] = set()
```

Comment:
```suggestion
    summary: dict[str, dict[str, Any]]
    current_packages: dict[str, dict[str, Any]]
```

I think it's wise to actually just keep the list of package tracking number of the last update in a set in `async_setup_entry` in `sensor.py`. This would make the coordinator less responsible for creating and removing entities. (and it concentrates the logic that we can remove in a few months)